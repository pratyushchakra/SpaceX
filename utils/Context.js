import React, { useState, useContext, useEffect } from 'react';
import createContainer from 'constate';

const nameHOC = (Component, suffix = '') => {
  return `${Component.originalName ||
    Component.displayName ||
    Component.name ||
    'Component'}${suffix}`;
};

const formatData = (allLaunchesData = []) => {
  return allLaunchesData.map(v => {
    return {
      ...v,
      isLandingTrue: v.rocket.first_stage.cores[v.rocket.first_stage.cores.length - 1].land_success
    }
  })
}

function useUserContext({ data: initialData }) {
  const [allLaunchesData, setLaunchData] = useState(initialData);
  const [filterData, setFilterData] = useState({
    launch: null, landing: null, year: ''
  });
  return {
    launchConfig: {
      allLaunches: [
        ...formatData(allLaunchesData)
      ],
      setLaunchData
    },
    filterConfig: {
      filterData,
      setFilterData
    }
  }
}

const LaunchInfoContext = createContainer(useUserContext);

const withContext = (WrappedComponent) => {
  const ComponentWithData = (props) => {
    const store = useContext(LaunchInfoContext.Context);
    return <WrappedComponent {...props} store={store} />;
  };

  ComponentWithData.originalName = nameHOC(WrappedComponent);
  ComponentWithData.displayName = nameHOC(
    WrappedComponent
  );
  return ComponentWithData;
};

export default LaunchInfoContext;
export { withContext };
