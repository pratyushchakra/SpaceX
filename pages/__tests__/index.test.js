import Enzyme, { shallow, mount } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

import Component from '../';
import Context from '../../utils/Context';
import mockData from '../../utils/mock';
import moxios from 'moxios';
beforeEach(() => {
 
})
beforeEach(() => {
    moxios.install()
    moxios.stubRequest('https://api.spacexdata.com/v3/launches?limit=100', {
        status: 200,
        response: mockData
    });
});

afterEach(() => {
    moxios.uninstall()
})

Enzyme.configure({ adapter: new Adapter() });

describe('spacex page', () => {
    it('renders the spaceX page as expected', async () => {
        const props = await Component.getInitialProps()
        const wrapper = mount(
            <Component {...props} />
        );
        expect(wrapper.debug()).toMatchSnapshot();
    });
    it('renders child component with proper data', async () => {
        const props = await Component.getInitialProps()
        const wrapper = mount(
            <Component {...props} />
        );
        expect(wrapper.find('Pod').length).toBe(2);
        expect(wrapper.find('Filter')).toBeTruthy();
    });
    it('does not render child components when service return no data', () => {
        const wrapper = mount(
            <Component />
        );
        expect(wrapper.debug()).toMatchSnapshot();
        expect(wrapper.find('Pod').length).toBe(0);
    });
    it('renders app smoothly even when service fails', () => {
        moxios.install()
        moxios.stubRequest('https://api.spacexdata.com/v3/launches?limit=100', {
            status: 500,
            response: undefined
        });
        const wrapper = mount(
            <Component />
        );
        expect(wrapper.debug()).toMatchSnapshot();
        expect(wrapper.find('Pod').length).toBe(0);
    })
});