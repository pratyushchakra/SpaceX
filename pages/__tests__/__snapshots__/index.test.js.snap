// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`spacex page does not render child components when service return no data 1`] = `
"<SpaceXProgram allLaunches={{...}}>
  <useUserContext.Provider data={{...}}>
    <Homepage>
      <div className=\\"jsx-2218827264\\">
        <header className=\\"jsx-2218827264\\">
          <h1 className=\\"jsx-2218827264\\">
            SpaceX Launch Program
          </h1>
        </header>
        <section className=\\"jsx-2218827264\\">
          <Portfolio>
            <div className=\\"jsx-1419685100 page-layout\\">
              <div className=\\"jsx-1419685100 filters-container\\">
                <Filters>
                  <Filters store={{...}}>
                    <p className=\\"jsx-1376872141 filters\\">
                      Filters
                    </p>
                    <div className=\\"jsx-1376872141 date-filter\\">
                      <p className=\\"jsx-1376872141 filter-name-text\\">
                        Launch Year
                      </p>
                      <div className=\\"jsx-1376872141 border-bottom\\" />
                      <div className=\\"jsx-1376872141 date-btns\\">
                        <div className=\\"jsx-1376872141 button-wrapper\\">
                          <button id=\\"year-btn-2008\\" onClick={[Function (anonymous)]} className=\\"jsx-1376872141 \\">
                            2008
                          </button>
                        </div>
                        <div className=\\"jsx-1376872141 button-wrapper\\">
                          <button id=\\"year-btn-2009\\" onClick={[Function (anonymous)]} className=\\"jsx-1376872141 \\">
                            2009
                          </button>
                        </div>
                        <div className=\\"jsx-1376872141 button-wrapper\\">
                          <button id=\\"year-btn-2010\\" onClick={[Function (anonymous)]} className=\\"jsx-1376872141 \\">
                            2010
                          </button>
                        </div>
                        <div className=\\"jsx-1376872141 button-wrapper\\">
                          <button id=\\"year-btn-2011\\" onClick={[Function (anonymous)]} className=\\"jsx-1376872141 \\">
                            2011
                          </button>
                        </div>
                        <div className=\\"jsx-1376872141 button-wrapper\\">
                          <button id=\\"year-btn-2012\\" onClick={[Function (anonymous)]} className=\\"jsx-1376872141 \\">
                            2012
                          </button>
                        </div>
                        <div className=\\"jsx-1376872141 button-wrapper\\">
                          <button id=\\"year-btn-2013\\" onClick={[Function (anonymous)]} className=\\"jsx-1376872141 \\">
                            2013
                          </button>
                        </div>
                        <div className=\\"jsx-1376872141 button-wrapper\\">
                          <button id=\\"year-btn-2014\\" onClick={[Function (anonymous)]} className=\\"jsx-1376872141 \\">
                            2014
                          </button>
                        </div>
                        <div className=\\"jsx-1376872141 button-wrapper\\">
                          <button id=\\"year-btn-2015\\" onClick={[Function (anonymous)]} className=\\"jsx-1376872141 \\">
                            2015
                          </button>
                        </div>
                        <div className=\\"jsx-1376872141 button-wrapper\\">
                          <button id=\\"year-btn-2016\\" onClick={[Function (anonymous)]} className=\\"jsx-1376872141 \\">
                            2016
                          </button>
                        </div>
                        <div className=\\"jsx-1376872141 button-wrapper\\">
                          <button id=\\"year-btn-2017\\" onClick={[Function (anonymous)]} className=\\"jsx-1376872141 \\">
                            2017
                          </button>
                        </div>
                        <div className=\\"jsx-1376872141 button-wrapper\\">
                          <button id=\\"year-btn-2018\\" onClick={[Function (anonymous)]} className=\\"jsx-1376872141 \\">
                            2018
                          </button>
                        </div>
                        <div className=\\"jsx-1376872141 button-wrapper\\">
                          <button id=\\"year-btn-2019\\" onClick={[Function (anonymous)]} className=\\"jsx-1376872141 \\">
                            2019
                          </button>
                        </div>
                        <div className=\\"jsx-1376872141 button-wrapper\\">
                          <button id=\\"year-btn-2020\\" onClick={[Function (anonymous)]} className=\\"jsx-1376872141 \\">
                            2020
                          </button>
                        </div>
                        <div className=\\"jsx-1376872141 button-wrapper\\">
                          <button id=\\"year-btn-2021\\" onClick={[Function (anonymous)]} className=\\"jsx-1376872141 \\">
                            2021
                          </button>
                        </div>
                        <div className=\\"jsx-1376872141 button-wrapper\\">
                          <button id=\\"year-btn-2022\\" onClick={[Function (anonymous)]} className=\\"jsx-1376872141 \\">
                            2022
                          </button>
                        </div>
                      </div>
                    </div>
                    <div className=\\"jsx-1376872141 launch-filter\\">
                      <p className=\\"jsx-1376872141 filter-name-text\\">
                        Successful Launch
                      </p>
                      <div className=\\"jsx-1376872141 border-bottom\\" />
                      <div className=\\"jsx-1376872141 date-btns\\">
                        <div className=\\"jsx-1376872141 button-wrapper\\">
                          <button onClick={[Function (anonymous)]} className=\\"jsx-1376872141 \\">
                            True
                          </button>
                        </div>
                        <div className=\\"jsx-1376872141 button-wrapper\\">
                          <button onClick={[Function (anonymous)]} className=\\"jsx-1376872141 \\">
                            False
                          </button>
                        </div>
                      </div>
                    </div>
                    <div className=\\"jsx-1376872141 landing-filter\\">
                      <p className=\\"jsx-1376872141 filter-name-text\\">
                        Successfull Landing
                      </p>
                      <div className=\\"jsx-1376872141 border-bottom\\" />
                      <div className=\\"jsx-1376872141 date-btns\\">
                        <div className=\\"jsx-1376872141 button-wrapper\\">
                          <button onClick={[Function (anonymous)]} className=\\"jsx-1376872141 \\">
                            True
                          </button>
                        </div>
                        <div className=\\"jsx-1376872141 button-wrapper\\">
                          <button onClick={[Function (anonymous)]} className=\\"jsx-1376872141 \\">
                            False
                          </button>
                        </div>
                      </div>
                    </div>
                    <JSXStyle id=\\"1376872141\\" />
                  </Filters>
                </Filters>
              </div>
              <div className=\\"jsx-1419685100 pods-container\\">
                <Pods>
                  <Pods store={{...}}>
                    <JSXStyle id=\\"1350581516\\" />
                  </Pods>
                </Pods>
              </div>
              <JSXStyle id=\\"1419685100\\" />
            </div>
          </Portfolio>
        </section>
        <footer className=\\"jsx-2218827264\\">
          <b className=\\"jsx-2218827264\\">
            Developed By:
          </b>
          <p className=\\"jsx-2218827264\\">
            Pratyush Chakra
          </p>
        </footer>
        <JSXStyle id=\\"2218827264\\" />
      </div>
    </Homepage>
  </useUserContext.Provider>
</SpaceXProgram>"
`;

exports[`spacex page renders app smoothly even when service fails 1`] = `
"<SpaceXProgram allLaunches={{...}}>
  <useUserContext.Provider data={{...}}>
    <Homepage>
      <div className=\\"jsx-2218827264\\">
        <header className=\\"jsx-2218827264\\">
          <h1 className=\\"jsx-2218827264\\">
            SpaceX Launch Program
          </h1>
        </header>
        <section className=\\"jsx-2218827264\\">
          <Portfolio>
            <div className=\\"jsx-1419685100 page-layout\\">
              <div className=\\"jsx-1419685100 filters-container\\">
                <Filters>
                  <Filters store={{...}}>
                    <p className=\\"jsx-1376872141 filters\\">
                      Filters
                    </p>
                    <div className=\\"jsx-1376872141 date-filter\\">
                      <p className=\\"jsx-1376872141 filter-name-text\\">
                        Launch Year
                      </p>
                      <div className=\\"jsx-1376872141 border-bottom\\" />
                      <div className=\\"jsx-1376872141 date-btns\\">
                        <div className=\\"jsx-1376872141 button-wrapper\\">
                          <button id=\\"year-btn-2008\\" onClick={[Function (anonymous)]} className=\\"jsx-1376872141 \\">
                            2008
                          </button>
                        </div>
                        <div className=\\"jsx-1376872141 button-wrapper\\">
                          <button id=\\"year-btn-2009\\" onClick={[Function (anonymous)]} className=\\"jsx-1376872141 \\">
                            2009
                          </button>
                        </div>
                        <div className=\\"jsx-1376872141 button-wrapper\\">
                          <button id=\\"year-btn-2010\\" onClick={[Function (anonymous)]} className=\\"jsx-1376872141 \\">
                            2010
                          </button>
                        </div>
                        <div className=\\"jsx-1376872141 button-wrapper\\">
                          <button id=\\"year-btn-2011\\" onClick={[Function (anonymous)]} className=\\"jsx-1376872141 \\">
                            2011
                          </button>
                        </div>
                        <div className=\\"jsx-1376872141 button-wrapper\\">
                          <button id=\\"year-btn-2012\\" onClick={[Function (anonymous)]} className=\\"jsx-1376872141 \\">
                            2012
                          </button>
                        </div>
                        <div className=\\"jsx-1376872141 button-wrapper\\">
                          <button id=\\"year-btn-2013\\" onClick={[Function (anonymous)]} className=\\"jsx-1376872141 \\">
                            2013
                          </button>
                        </div>
                        <div className=\\"jsx-1376872141 button-wrapper\\">
                          <button id=\\"year-btn-2014\\" onClick={[Function (anonymous)]} className=\\"jsx-1376872141 \\">
                            2014
                          </button>
                        </div>
                        <div className=\\"jsx-1376872141 button-wrapper\\">
                          <button id=\\"year-btn-2015\\" onClick={[Function (anonymous)]} className=\\"jsx-1376872141 \\">
                            2015
                          </button>
                        </div>
                        <div className=\\"jsx-1376872141 button-wrapper\\">
                          <button id=\\"year-btn-2016\\" onClick={[Function (anonymous)]} className=\\"jsx-1376872141 \\">
                            2016
                          </button>
                        </div>
                        <div className=\\"jsx-1376872141 button-wrapper\\">
                          <button id=\\"year-btn-2017\\" onClick={[Function (anonymous)]} className=\\"jsx-1376872141 \\">
                            2017
                          </button>
                        </div>
                        <div className=\\"jsx-1376872141 button-wrapper\\">
                          <button id=\\"year-btn-2018\\" onClick={[Function (anonymous)]} className=\\"jsx-1376872141 \\">
                            2018
                          </button>
                        </div>
                        <div className=\\"jsx-1376872141 button-wrapper\\">
                          <button id=\\"year-btn-2019\\" onClick={[Function (anonymous)]} className=\\"jsx-1376872141 \\">
                            2019
                          </button>
                        </div>
                        <div className=\\"jsx-1376872141 button-wrapper\\">
                          <button id=\\"year-btn-2020\\" onClick={[Function (anonymous)]} className=\\"jsx-1376872141 \\">
                            2020
                          </button>
                        </div>
                        <div className=\\"jsx-1376872141 button-wrapper\\">
                          <button id=\\"year-btn-2021\\" onClick={[Function (anonymous)]} className=\\"jsx-1376872141 \\">
                            2021
                          </button>
                        </div>
                        <div className=\\"jsx-1376872141 button-wrapper\\">
                          <button id=\\"year-btn-2022\\" onClick={[Function (anonymous)]} className=\\"jsx-1376872141 \\">
                            2022
                          </button>
                        </div>
                      </div>
                    </div>
                    <div className=\\"jsx-1376872141 launch-filter\\">
                      <p className=\\"jsx-1376872141 filter-name-text\\">
                        Successful Launch
                      </p>
                      <div className=\\"jsx-1376872141 border-bottom\\" />
                      <div className=\\"jsx-1376872141 date-btns\\">
                        <div className=\\"jsx-1376872141 button-wrapper\\">
                          <button onClick={[Function (anonymous)]} className=\\"jsx-1376872141 \\">
                            True
                          </button>
                        </div>
                        <div className=\\"jsx-1376872141 button-wrapper\\">
                          <button onClick={[Function (anonymous)]} className=\\"jsx-1376872141 \\">
                            False
                          </button>
                        </div>
                      </div>
                    </div>
                    <div className=\\"jsx-1376872141 landing-filter\\">
                      <p className=\\"jsx-1376872141 filter-name-text\\">
                        Successfull Landing
                      </p>
                      <div className=\\"jsx-1376872141 border-bottom\\" />
                      <div className=\\"jsx-1376872141 date-btns\\">
                        <div className=\\"jsx-1376872141 button-wrapper\\">
                          <button onClick={[Function (anonymous)]} className=\\"jsx-1376872141 \\">
                            True
                          </button>
                        </div>
                        <div className=\\"jsx-1376872141 button-wrapper\\">
                          <button onClick={[Function (anonymous)]} className=\\"jsx-1376872141 \\">
                            False
                          </button>
                        </div>
                      </div>
                    </div>
                    <JSXStyle id=\\"1376872141\\" />
                  </Filters>
                </Filters>
              </div>
              <div className=\\"jsx-1419685100 pods-container\\">
                <Pods>
                  <Pods store={{...}}>
                    <JSXStyle id=\\"1350581516\\" />
                  </Pods>
                </Pods>
              </div>
              <JSXStyle id=\\"1419685100\\" />
            </div>
          </Portfolio>
        </section>
        <footer className=\\"jsx-2218827264\\">
          <b className=\\"jsx-2218827264\\">
            Developed By:
          </b>
          <p className=\\"jsx-2218827264\\">
            Pratyush Chakra
          </p>
        </footer>
        <JSXStyle id=\\"2218827264\\" />
      </div>
    </Homepage>
  </useUserContext.Provider>
</SpaceXProgram>"
`;

exports[`spacex page renders the spaceX page as expected 1`] = `
"<SpaceXProgram allLaunches={{...}}>
  <useUserContext.Provider data={{...}}>
    <Homepage>
      <div className=\\"jsx-2218827264\\">
        <header className=\\"jsx-2218827264\\">
          <h1 className=\\"jsx-2218827264\\">
            SpaceX Launch Program
          </h1>
        </header>
        <section className=\\"jsx-2218827264\\">
          <Portfolio>
            <div className=\\"jsx-1419685100 page-layout\\">
              <div className=\\"jsx-1419685100 filters-container\\">
                <Filters>
                  <Filters store={{...}}>
                    <p className=\\"jsx-1376872141 filters\\">
                      Filters
                    </p>
                    <div className=\\"jsx-1376872141 date-filter\\">
                      <p className=\\"jsx-1376872141 filter-name-text\\">
                        Launch Year
                      </p>
                      <div className=\\"jsx-1376872141 border-bottom\\" />
                      <div className=\\"jsx-1376872141 date-btns\\">
                        <div className=\\"jsx-1376872141 button-wrapper\\">
                          <button id=\\"year-btn-2008\\" onClick={[Function (anonymous)]} className=\\"jsx-1376872141 \\">
                            2008
                          </button>
                        </div>
                        <div className=\\"jsx-1376872141 button-wrapper\\">
                          <button id=\\"year-btn-2009\\" onClick={[Function (anonymous)]} className=\\"jsx-1376872141 \\">
                            2009
                          </button>
                        </div>
                        <div className=\\"jsx-1376872141 button-wrapper\\">
                          <button id=\\"year-btn-2010\\" onClick={[Function (anonymous)]} className=\\"jsx-1376872141 \\">
                            2010
                          </button>
                        </div>
                        <div className=\\"jsx-1376872141 button-wrapper\\">
                          <button id=\\"year-btn-2011\\" onClick={[Function (anonymous)]} className=\\"jsx-1376872141 \\">
                            2011
                          </button>
                        </div>
                        <div className=\\"jsx-1376872141 button-wrapper\\">
                          <button id=\\"year-btn-2012\\" onClick={[Function (anonymous)]} className=\\"jsx-1376872141 \\">
                            2012
                          </button>
                        </div>
                        <div className=\\"jsx-1376872141 button-wrapper\\">
                          <button id=\\"year-btn-2013\\" onClick={[Function (anonymous)]} className=\\"jsx-1376872141 \\">
                            2013
                          </button>
                        </div>
                        <div className=\\"jsx-1376872141 button-wrapper\\">
                          <button id=\\"year-btn-2014\\" onClick={[Function (anonymous)]} className=\\"jsx-1376872141 \\">
                            2014
                          </button>
                        </div>
                        <div className=\\"jsx-1376872141 button-wrapper\\">
                          <button id=\\"year-btn-2015\\" onClick={[Function (anonymous)]} className=\\"jsx-1376872141 \\">
                            2015
                          </button>
                        </div>
                        <div className=\\"jsx-1376872141 button-wrapper\\">
                          <button id=\\"year-btn-2016\\" onClick={[Function (anonymous)]} className=\\"jsx-1376872141 \\">
                            2016
                          </button>
                        </div>
                        <div className=\\"jsx-1376872141 button-wrapper\\">
                          <button id=\\"year-btn-2017\\" onClick={[Function (anonymous)]} className=\\"jsx-1376872141 \\">
                            2017
                          </button>
                        </div>
                        <div className=\\"jsx-1376872141 button-wrapper\\">
                          <button id=\\"year-btn-2018\\" onClick={[Function (anonymous)]} className=\\"jsx-1376872141 \\">
                            2018
                          </button>
                        </div>
                        <div className=\\"jsx-1376872141 button-wrapper\\">
                          <button id=\\"year-btn-2019\\" onClick={[Function (anonymous)]} className=\\"jsx-1376872141 \\">
                            2019
                          </button>
                        </div>
                        <div className=\\"jsx-1376872141 button-wrapper\\">
                          <button id=\\"year-btn-2020\\" onClick={[Function (anonymous)]} className=\\"jsx-1376872141 \\">
                            2020
                          </button>
                        </div>
                        <div className=\\"jsx-1376872141 button-wrapper\\">
                          <button id=\\"year-btn-2021\\" onClick={[Function (anonymous)]} className=\\"jsx-1376872141 \\">
                            2021
                          </button>
                        </div>
                        <div className=\\"jsx-1376872141 button-wrapper\\">
                          <button id=\\"year-btn-2022\\" onClick={[Function (anonymous)]} className=\\"jsx-1376872141 \\">
                            2022
                          </button>
                        </div>
                      </div>
                    </div>
                    <div className=\\"jsx-1376872141 launch-filter\\">
                      <p className=\\"jsx-1376872141 filter-name-text\\">
                        Successful Launch
                      </p>
                      <div className=\\"jsx-1376872141 border-bottom\\" />
                      <div className=\\"jsx-1376872141 date-btns\\">
                        <div className=\\"jsx-1376872141 button-wrapper\\">
                          <button onClick={[Function (anonymous)]} className=\\"jsx-1376872141 \\">
                            True
                          </button>
                        </div>
                        <div className=\\"jsx-1376872141 button-wrapper\\">
                          <button onClick={[Function (anonymous)]} className=\\"jsx-1376872141 \\">
                            False
                          </button>
                        </div>
                      </div>
                    </div>
                    <div className=\\"jsx-1376872141 landing-filter\\">
                      <p className=\\"jsx-1376872141 filter-name-text\\">
                        Successfull Landing
                      </p>
                      <div className=\\"jsx-1376872141 border-bottom\\" />
                      <div className=\\"jsx-1376872141 date-btns\\">
                        <div className=\\"jsx-1376872141 button-wrapper\\">
                          <button onClick={[Function (anonymous)]} className=\\"jsx-1376872141 \\">
                            True
                          </button>
                        </div>
                        <div className=\\"jsx-1376872141 button-wrapper\\">
                          <button onClick={[Function (anonymous)]} className=\\"jsx-1376872141 \\">
                            False
                          </button>
                        </div>
                      </div>
                    </div>
                    <JSXStyle id=\\"1376872141\\" />
                  </Filters>
                </Filters>
              </div>
              <div className=\\"jsx-1419685100 pods-container\\">
                <Pods>
                  <Pods store={{...}}>
                    <Pod launch={{...}}>
                      <div className=\\"jsx-1350581516 pod\\">
                        <div className=\\"jsx-1350581516 image-container\\">
                          <img src=\\"https://images2.imgbox.com/ff/81/EOWojaSj_o.png\\" width=\\"200\\" className=\\"jsx-1350581516\\" />
                        </div>
                        <p className=\\"jsx-1350581516 mission-name\\">
                          CRS-3
                           #
                          14
                        </p>
                        <div className=\\"jsx-1350581516 text-holder\\">
                          <p className=\\"jsx-1350581516 bold-text\\">
                            Mission IDs:
                          </p>
                          <p className=\\"jsx-1350581516\\">
                            EE86F74
                          </p>
                        </div>
                        <div className=\\"jsx-1350581516 text-holder\\">
                          <p className=\\"jsx-1350581516 bold-text\\">
                            Launch Year:
                          </p>
                          <p className=\\"jsx-1350581516\\">
                            2014
                          </p>
                        </div>
                        <div className=\\"jsx-1350581516 text-holder\\">
                          <p className=\\"jsx-1350581516 bold-text\\">
                            Successful Launch:
                          </p>
                          <p className=\\"jsx-1350581516\\">
                            True
                          </p>
                        </div>
                        <div className=\\"jsx-1350581516 text-holder\\">
                          <p className=\\"jsx-1350581516 bold-text\\">
                            Successful Landing:
                          </p>
                          <p className=\\"jsx-1350581516\\">
                            True
                          </p>
                        </div>
                        <JSXStyle id=\\"1350581516\\" />
                      </div>
                    </Pod>
                    <Pod launch={{...}}>
                      <div className=\\"jsx-1350581516 pod\\">
                        <div className=\\"jsx-1350581516 image-container\\">
                          <img src=\\"https://images2.imgbox.com/4d/ed/CHXoRaSP_o.png\\" width=\\"200\\" className=\\"jsx-1350581516\\" />
                        </div>
                        <p className=\\"jsx-1350581516 mission-name\\">
                          OG-2 Mission 1
                           #
                          15
                        </p>
                        <div className=\\"jsx-1350581516 text-holder\\">
                          <p className=\\"jsx-1350581516 bold-text\\">
                            Mission IDs:
                          </p>
                          <p className=\\"jsx-1350581516\\">
                            CE91D46
                          </p>
                        </div>
                        <div className=\\"jsx-1350581516 text-holder\\">
                          <p className=\\"jsx-1350581516 bold-text\\">
                            Launch Year:
                          </p>
                          <p className=\\"jsx-1350581516\\">
                            2014
                          </p>
                        </div>
                        <div className=\\"jsx-1350581516 text-holder\\">
                          <p className=\\"jsx-1350581516 bold-text\\">
                            Successful Launch:
                          </p>
                          <p className=\\"jsx-1350581516\\">
                            True
                          </p>
                        </div>
                        <div className=\\"jsx-1350581516 text-holder\\">
                          <p className=\\"jsx-1350581516 bold-text\\">
                            Successful Landing:
                          </p>
                          <p className=\\"jsx-1350581516\\">
                            True
                          </p>
                        </div>
                        <JSXStyle id=\\"1350581516\\" />
                      </div>
                    </Pod>
                    <JSXStyle id=\\"1350581516\\" />
                  </Pods>
                </Pods>
              </div>
              <JSXStyle id=\\"1419685100\\" />
            </div>
          </Portfolio>
        </section>
        <footer className=\\"jsx-2218827264\\">
          <b className=\\"jsx-2218827264\\">
            Developed By:
          </b>
          <p className=\\"jsx-2218827264\\">
            Pratyush Chakra
          </p>
        </footer>
        <JSXStyle id=\\"2218827264\\" />
      </div>
    </Homepage>
  </useUserContext.Provider>
</SpaceXProgram>"
`;
